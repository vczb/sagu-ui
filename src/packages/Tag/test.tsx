import React from 'react'
import { render, screen } from '../../utils/testUtils'

import Tag from '.'

describe('<Tag/>', () => {
  it('should render the tag value given', () => {
    render(<Tag value="coffee" />)

    expect(screen.getByText('coffee')).toBeInTheDocument()
  })

  it('should render tag with uppercase content', async () => {
    const dom = render(<Tag value="coffee" isUppercase={true} />)

    expect(dom.getByText(/COFFEE/i)).toBeInTheDocument()
  })

  it('should render slashed bold tag', async () => {
    const { container } = render(
      <Tag value="coffee" isUppercase={true} isBold isSlashed />
    )

    expect(container).toMatchSnapshot()
  })
})
