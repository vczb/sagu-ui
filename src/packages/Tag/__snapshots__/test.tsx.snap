// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag/> should render slashed bold tag 1`] = `
.c0 {
  padding: 0.4rem 0.8rem;
  background: #1C5D9F;
  color: #F8F7FB;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 400;
  text-transform: uppercase;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  font-weight: 700;
}

.c0:hover {
  background: #1C5D9F;
}

<div>
  <div
    class="c0"
  >
    coffee
  </div>
</div>
`;
